rosidl_typesupport_opendds_cpp/resource/msg__type_support.cpp.em

```diff
@@ -313,7 +325,24 @@ to_message__@(message.structure.namespaced_type.name)(
   }
 
   OpenDDS::DCPS::Message_Block_Ptr b(new ACE_Message_Block(cdr_stream->buffer_length));
-  OpenDDS::DCPS::Serializer deserializer(b.get(), false);
+  memcpy(b->wr_ptr(), cdr_stream->buffer, cdr_stream->buffer_length);
+  b->wr_ptr(cdr_stream->buffer_length);
+
+  OpenDDS::DCPS::Serializer deserializer(b.get(), false, OpenDDS::DCPS::Serializer::ALIGN_CDR);
+
+  const size_t header_size = 4;
```

Comment:
Is this size defined elsewhere?

Reply:
I didn't find an explicit definition that I could use.

Reply:
OpenDDS uses a different approach because the 1st two bytes and the last two bytes are treated separately.  For the usage here in ROSIDL TS I think it's fine.  The next iteration of XTypes in OpenDDS (not in master yet) will have a better way of handling it.